aten/src/ATen/native/cpu/DistanceOpsKernel.cpp

```diff
@@ -155,6 +155,55 @@ struct PDist {
     }
   }
 
+    template <typename F>
+    static void run_cdist_parallel(Tensor& result, const Tensor& t1, const Tensor& t2, const scalar_t p) {
+        const scalar_t * const t1_start = t1.data<scalar_t>();
+        const scalar_t * const t2_start = t2.data<scalar_t>();
+        int64_t r1 = t1.size(-2);
+        int64_t r2 = t2.size(-2);
+        int64_t m = t1.size(-1);
+
+        scalar_t * const res_start = result.data<scalar_t>();
+        int64_t total = r1 * r2;
+        const Vec pvec(p);
+
+        parallel_for(0, total, internal::GRAIN_SIZE / (16 * m), [=, &pvec](int64_t start, int64_t end) {
+            scalar_t * res = res_start + start;
+            const scalar_t * const res_end = res_start + end;
+
+            int64_t k = 0;
+            while (res != res_end) {
+                int64_t i = k / r2;
```

Comment:
this looks like you don't calculate the correct elements -- the pointer positions are correct but i, j, k aren't updated based on start, end.

You can view this by trying a big input on your test data.  For example, try (1000, 1000) by (1000, 1000), but I'm guessing anything that parallelizes will break.